features/environment.py

```diff
@@ -103,7 +103,7 @@ def _setup_minimal_env(context):
             "pip",
             "install",
             "-U",
-            "pip>=21.2",
+            "pip>=21.2, <23.2",
```

Comment:
Maybe we can cap only on Python 3.7, see my comment in https://github.com/kedro-org/kedro/pull/2816#issuecomment-1650426611